server/routes/link/RequestLinkController.ts

```diff
@@ -1,10 +1,25 @@
 import { Request, Response } from 'express'
 import RequestLinkView from './RequestLinkView'
+import validate from './RequestLinkValidator'
 
 export default class RequestLinkController {
-  async getRequestLinkView(req: Request, res: Response) {
+  async getRequestLinkView(req: Request, res: Response): Promise<void> {
     const view = new RequestLinkView(req.session?.requestLinkForm || {}, req.flash('errors'))
 
-    res.clearCookie('create_barcode_token').render('pages/link/requestLink', view.renderArgs)
+    res.clearCookie('create_barcode_token')
+    return res.render('pages/link/requestLink', { ...view.renderArgs })
+  }
+
+  async submitLinkRequest(req: Request, res: Response): Promise<void> {
+    req.session.requestLinkForm = { ...req.body }
+    if (!validate(req.session.requestLinkForm, req)) {
+      return res.redirect('request-link')
+    }
```

Comment:
Is it worth a small unit test to capture this bit of logic - though it is covered by the integration tests...

Reply:
As you say, it is covered by the integration tests. The cypress test asserts the user ends up on the correct page, but not the detail of how we got them there. A unit test would test the implementation (ie. whether we did a redirect, or re-rendering the original view etc.) I think the important thing in this case is the 'what' rather than the 'how', so I'd be tempted to leave it as a cypress test, but happy to be challenged on this